### PR TITLE
subject list with filter

### PR DESCRIPTION
**Pull request by Tusharmahajan12**

### CODE REVIEW

File: src/src/app.module.ts [hunk 1]
```diff
@@ -39,6 +40,7 @@ import { AttendancesModule } from './attendances/attendances.module';
     UploadFileModule,
     AttendancesModule,
     KeycloakModule,
+    SubjectslistModule,
```
Comment (manojLondhe) [inline src/src/app.module.ts:43]: rename Subjectslist to Subjects everywhere

File: src/src/subjectslist/subjectslist.controller.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+} from '@nestjs/common';
+import { SubjectslistService } from './subjectslist.service';
+import { CreateSubjectslistDto } from './dto/create-subjectslist.dto';
+import { UpdateSubjectslistDto } from './dto/update-subjectslist.dto';
+
+@Controller('subjectslist')
+export class SubjectslistController {
+  constructor(private readonly subjectslistService: SubjectslistService) {}
+
+  @Post('/create')
```
Comment (manojLondhe) [inline src/src/subjectslist/subjectslist.controller.ts:18]: make all routes protected for - add /edit / delete

or comment all excecpt the for getting list of subjects

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/src/auth/auth.service.ts [hunk 3]
```diff
@@ -400,25 +400,18 @@ export class AuthService {
         };
         const token = await this.keycloakService.getAdminKeycloakToken(query, 'master')
         if (token?.access_token) {
-
-            const findUsername = await this.keycloakService.findUser(data_to_create_user, token.access_token)
+            const findUsername = await this.keycloakService.findUser(username, token?.access_token)
```
Comment (sagarkoshti1990) [inline src/src/auth/auth.service.ts:403]: move this in `if (body.role === 'beneficiaries')`

File: src/src/services/keycloak/keycloak.service.ts [hunk 4]
```diff
@@ -112,7 +112,7 @@ export class KeycloakService {
     
 
     public async registerUser(data, token) {
-        console.log("inside registerUser")
+        console.log("inside registerUser", data)
```
Comment (sagarkoshti1990) [inline src/src/services/keycloak/keycloak.service.ts:115]: please remove console log. if not needed.

File: src/src/services/keycloak/keycloak.service.ts [hunk 5]
```diff
@@ -146,7 +146,7 @@ export class KeycloakService {
     public async findUser(data, token) {
         console.log("inside findUser", data)
 
-        const url = `${this.keycloak_url}/admin/realms/eg-sso/users?username=${data.username}`;
+        const url = `${this.keycloak_url}/admin/realms/eg-sso/users?username=${data}`;
```
Comment (sagarkoshti1990) [inline src/src/services/keycloak/keycloak.service.ts:149]:
```
const url = `${this.keycloak_url}/admin/realms/${this.<eg-sso>}/users?username=${data}`;
```

**Pull request closed without merging**